src/App.js

```diff
@@ -32,15 +33,69 @@ function App() {
       job: "github",
     },
   };
+  const [apiData, setApiData] = useState(UserData);
 
   const ContactIcons = [faLocationDot, faTwitter, faLink, faBuilding];
 
+  const searchHandler = async (username) => {
+    try {
+      const resUser = await fetch(`https://api.github.com/users/${username}`);
+      if (!resUser.ok) throw new Error("Problem getting user data");
+      console.log(resUser);
+      const data = await resUser.json();
+
+      const {
+        avatar_url,
+        name,
+        login,
+        created_at,
+        bio,
+        public_repos,
+        followers,
+        following,
+        location,
+        twitter_username,
+        blog,
+        company,
+      } = data;
+
+      setApiData({
+        profile: {
+          image: `${avatar_url}`,
+          name: `${name || "Not Available"}`,
+          handle: `${login || "Not Available"}`,
+          bio: `${bio || "This profile has no bio"}`,
+          date: `Joined ${
+            new Date(created_at).toLocaleString("en-US", {
+              day: "2-digit",
+              month: "short",
+              year: "numeric",
+            }) || "Not Available"
+          }`,
+        },
+        account: {
+          repo: `${public_repos || "-"}`,
+          followers: `${followers || "-"}`,
+          following: `${following || "-"}`,
+        },
+        contact: {
+          location: `${location || "Not Available"}`,
+          twitter: `${twitter_username || "Not Available"}`,
+          website: `${blog || "Not Available"}`,
+          job: `${company || "Not Available"}`,
+        },
+      });
+    } catch (err) {
+      console.error(`An error occurred: ${err}`);
+    }
+  };
+
   return (
     <div className={styles.body}>
       <div className={styles.displayCard}>
         <Header />
-        <SearchBar icon={faMagnifyingGlass} />
-        <UserDisplay icons={ContactIcons} userData={UserData} />
+        <SearchBar icon={faMagnifyingGlass} searchHandler={searchHandler} />
+        <UserDisplay icons={ContactIcons} userData={apiData} />
```

Comment:
The icons in these components are only used in those specific components. A good approach would be to import them in their specific components rather that passing them down as props. This reduces the responsibility to only the components that need the icons. So, import all `ContactIcons` in `UserDisplay` and `magnifyingGlass` inside the `SearchBar` component and then remove them from props.